x/name/types/msgs.go

```diff
@@ -101,3 +103,34 @@ func (msg MsgDeleteNameRequest) GetSigners() []sdk.AccAddress {
 	}
 	return []sdk.AccAddress{addr}
 }
+
+// NewMsgModifyNameRequest modifies an existing name record
+func NewMsgModifyNameRequest(authority string, name string, owner sdk.AccAddress, restricted bool) *MsgModifyNameRequest {
+	return &MsgModifyNameRequest{
+		Authority: authority,
+		Record:    NewNameRecord(name, owner, restricted),
+	}
+}
+
+// ValidateBasic runs stateless validation checks on the message.
+func (msg MsgModifyNameRequest) ValidateBasic() error {
+	if strings.TrimSpace(msg.Record.Name) == "" {
+		return fmt.Errorf("name cannot be empty")
+	}
+	if strings.TrimSpace(msg.Record.Address) == "" {
+		return fmt.Errorf("address cannot be empty")
+	}
```

Comment:
Should check if the `msg.Record.Address` is a valid Bech32 instead of just checking for empty/whitespace.
```suggestion
	if _, err := sdk. AccAddressFromBech32(msg.Record.Address); err != nil {
		return fmt.Errorf("invalid record address: %w", err)
	}
```
For the authority, though, it's okay to just do the empty string check since it's only ever used as a string.